src/bitcoind/actions.rs

```diff
@@ -319,9 +331,13 @@ fn poll_bitcoind(revaultd: &mut RevaultD, bitcoind: &BitcoinD) -> Result<(), Bit
             );
             // Note that it *might* have actually been confirmed during the last 30s, but it's not
             // a big deal to have it marked as unconfirmed for the next 30s..
-            db_unvault_deposit(&revaultd.db_file(), outpoint.txid.to_vec(), outpoint.vout)
-                // TODO: something like From<DbError> for BitcoindError ?
-                .map_err(|e| BitcoindError(format!("Database error: {}", e.to_string())))?;
+            db_unvault_deposit(
+                &revaultd.read().unwrap().db_file(),
```

Comment:
db_file typically has no chance to produce a race condition, revaultd.read().unwrap() is called here only to get access to the struct. Maybe It would be cleaner to add RWLock inside of the RevaultD struct for fields like `vaults` and `derivation_index_map` and handle the lock/unlock with methods to the struct

Reply:
Hmm good point. However not sure it would be `Send` then.. Also we would have to probably make it a member instead of a method.
Can we leave it for later studies of optimization (not even sure it's doable) ?

Reply:
> Can we leave it for later studies of optimization (not even sure it's doable) ?

Yes